jni_bindings/src/lib.rs

```diff
@@ -73,159 +69,226 @@ pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_00024Comp
     log::error!("Logging initialised from Rust");
 }
 
+/// Returns raw pointer to JavaResult
 #[no_mangle]
 pub extern "system" fn Java_org_phoenixframework_liveview_lib_Document_00024Companion_do_1parse<
     'local,
 >(
     mut env: JNIEnv<'local>,
     _: JClass<'local>,
     text: JString<'local>,
+    // raw pointer to JavaResult
 ) -> jlong {
     let text: String = env.get_string(&text).unwrap().into();
-    let result = match dom::Document::parse(text) {
+    let result = match Document::parse(text) {
         Ok(doc) => {
             let doc = Box::new(doc);
             JavaResult {
                 val: Box::into_raw(doc) as jlong,
-                error_msg: "".into(),
+                error_msg: String::new(),
             }
         }
         Err(err) => JavaResult {
             val: 0,
             error_msg: err.to_string(),
         },
     };
-    let ret = Box::new(result);
-    Box::into_raw(ret) as jlong
+    let result = Box::new(result);
+    Box::into_raw(result) as jlong
 }
 
 #[no_mangle]
-pub extern "system" fn Java_org_phoenixframework_liveview_lib_JavaResult_drop<'local>(
-    _env: JNIEnv<'local>,
+pub unsafe extern "system" fn Java_org_phoenixframework_liveview_lib_JavaResult_drop<'local>(
+    mut env: JNIEnv<'local>,
     _: JClass<'local>,
-    j_result: jlong,
+    // non-null raw pointer to JavaResult
```

Comment:
The `non-null` can't be true or there would be no need for the `this.is_null()` check below.

```suggestion
    // raw pointer to JavaResult
```